tensorhive/models/Reservation.py

```diff
@@ -54,35 +56,14 @@ def check_assertions(self):
     def duration(self):
         return self.ends_at - self.starts_at
 
-    @classmethod
-    def parsed_input_datetime(cls, value: str) -> Optional[datetime.datetime]:
-        client_datetime_format = '%Y-%m-%dT%H:%M:%S.%fZ'
-        try:
-            result = datetime.datetime.strptime(value, client_datetime_format)
-        except ValueError as e:
-            log.error(e)
-            raise
-        else:
-            return result
-
-    @classmethod
-    def parsed_output_datetime(cls, value: datetime.datetime) -> str:
-        display_datetime_format = '%Y-%m-%dT%H:%M:%S'
-        server_timezone = '+00:00'
-        return value.strftime(display_datetime_format) + server_timezone
-
     @hybrid_property
     def starts_at(self):
         return self._starts_at
 
     @starts_at.setter  # type: ignore
     def starts_at(self, value):
-        if isinstance(value, str):
-            self._starts_at = self.parsed_input_datetime(value)
-        elif isinstance(value, datetime.datetime):
-            self._starts_at = value
-        else:
-            self._starts_at = None
+        self._starts_at = DateUtils.try_parse_string(value)
+        if not self._starts_at:
```

Comment:
Do we expect any falsy value other than `None` here? If not we could replace this with the following code, so that it explicitly shows the intent of this check.

```suggestion
        if self._starts_at is not None:
```

Reply:
Nope - fixed ;)